# HG changeset patch
# Parent b735e618c2a860972197942f6680c941e3c9a67f
# User Gustavo Luiz Duarte <gustavold@linux.vnet.ibm.com>
Bug 997353 - Do not use compile-time page size for PowerPC.

The PowerPC architecture allows various memory page sizes, making it impossible
to predict the runtime page size.

diff -r 57f68296c350 memory/build/mozjemalloc.cpp
--- a/memory/build/mozjemalloc.cpp      Fri Sep 29 14:47:25 2017 -0700
+++ b/memory/build/mozjemalloc.cpp      Sat Sep 30 00:19:31 2017 +0200
@@ -387,13 +387,14 @@
  * compile-time for better performance, as opposed to determined at
  * runtime. Some platforms can have different page sizes at runtime
  * depending on kernel configuration, so they are opted out by default.
  * Debug builds are opted out too, for test coverage.
  */
 #ifndef MOZ_DEBUG
-#if !defined(__ia64__) && !defined(__sparc__) && !defined(__mips__) && !defined(__aarch64__)
+#if !defined(__ia64__) && !defined(__sparc__) && !defined(__mips__) && !defined(__aarch64__) && \
+    !defined(__powerpc__) && !defined(__powerpc64__)
 #define MALLOC_STATIC_PAGESIZE 1
 #endif
 #endif
 
 /* Various quantum-related settings. */
 

